File: android/src/main/kotlin/com/anggaaryas/gojek_courier/model/QosParam.kt
package com.anggaaryas.gojek_courier.model

import com.gojek.courier.QoS

class QosParam(value:String) : EnumParam<QoS>(value) {
    lateinit var qoS: QoS

    init {
        println(value)
        when(value){
            "ZERO" -> qoS = QoS.ZERO
            "ONE" -> qoS = QoS.ONE
            "TWO" -> qoS = QoS.TWO
        }
    }

    override fun build(): QoS {
        return qoS
    }
}

File: ios/Classes/model/QosParam.swift
//
//  QosParam.swift
//  gojek_courier
//
//  Created by Angga Arya Saputra on 14/07/22.
//

import Foundation
import CourierCore

class QosParam: EnumParam<QoS>{

    override func build() -> QoS {
        switch value {
        case "ZERO" :
            return QoS.zero
        case "ONE" :
            return QoS.one
        case "TWO" :
            return QoS.two
        default:
            fatalError("unknown qos : " + value)

        }
    }
}
